### PR TITLE
Code generator for protobuf specification

### PR DESCRIPTION
**Pull request by pablf**
/claim #1521

Generates ZIO Http endpoints from services with the Protobuf codec from ZIO Schema. Messages can be generated with scalapb. Technically, it could be already published as a sbt plugin using the project `sbt-zio-http-grpc`, but I think that it could make more sense to have a common sbt plugin with the OpenAPI generator.

### CODE REVIEW

File: zio-http-gen/src/main/scala/zio/http/gen/scala/Code.scala [hunk 1]
```diff
@@ -133,23 +133,25 @@ object Code {
...
+  object HeadersCode     { val empty: HeadersCode = HeadersCode(Nil)                             }
   final case class HeaderCode(name: String)
   final case class InCode(
     inType: String,
     name: Option[String],
     doc: Option[String],
+    streaming: Boolean,
   )
-  object InCode          { def apply(inType: String): InCode = InCode(inType, None, None) }
+  object InCode          { def apply(inType: String): InCode = InCode(inType, None, None, false) }
```
Comment (987Nabil) [inline zio-http-gen/src/main/scala/zio/http/gen/scala/Code.scala]: Please name the boolean params

File: zio-http-gen/src/main/scala/zio/http/gen/openapi/EndpointGen.scala [hunk 2]
```diff
@@ -283,6 +284,7 @@ final case class EndpointGen() {
             status = status,
             mediaType = Some("application/json"),
             doc = None,
+            false,
```
Comment (987Nabil) [inline zio-http-gen/src/main/scala/zio/http/gen/openapi/EndpointGen.scala]: Please name the boolean params

File: project/plugins.sbt [hunk 3]
```diff
@@ -12,3 +12,6 @@ addSbtPlugin("io.get-coursier"    % "sbt-shading"               % "2.1.4")
 addSbtPlugin("com.github.cb372"   % "sbt-explicit-dependencies" % "0.3.1")
 addSbtPlugin("org.scala-js"       % "sbt-scalajs"               % "1.16.0")
 addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject"  % "1.3.2")
+addSbtPlugin("com.thesamet"       % "sbt-protoc"                % "1.0.7")
+addSbtPlugin("com.thesamet"       % "sbt-protoc-gen-project"    % "0.1.8")
+libraryDependencies += "com.thesamet.scalapb" %% "compilerplugin" % "0.11.15"
```
Comment (987Nabil) [inline project/plugins.sbt:17]: I am not sure about this. Is this dependency to a compiler plugin? Or just a dependency to the sources of it?

**Review by 987Nabil — COMMENTED** [review]
I am not sure, I like the dependencies. I think the format of the protobuf files is not so complicated and we could maybe use RichTextCodec to parse it ourselves? We have already means to generate files anyway.
@jdegoes what is your opinion?

File: project/plugins.sbt (hunk 3 above)
Comment (pablf) [reply project/plugins.sbt:17]: It's a dependency for the build to be able to define the project `sbtZioHttpGrpcTests`.

**Comment by pablf** [comment]
We could convert the protobuf descriptor class to a `zio-http`-made descriptor within `sbt-zio-http-grpc` and move the protobuf dependencies to that project so they are not in `zio-http-gen`. Then, the final plugin could either parse itself the protobuf files or be published using scalapb.

**Review by jdegoes — APPROVED** [review]

**Pull request merged**